File: backend/src/main/java/com/example/hotel/service/DishService.java
package com.example.hotel.service;

import com.example.hotel.dto.DishDTO;
import java.util.List;

public interface DishService {
    List<DishDTO> getAllDishes();
    DishDTO getDishById(Long id);
    DishDTO createDish(DishDTO dishDTO);
    DishDTO updateDish(Long id, DishDTO dishDTO);
    void deleteDish(Long id);
    List<DishDTO> getDishesByCategory(String category);
}

File: src/services/dishService.js
// src/services/dishService.js
import axiosInstance from '../config/axios.config';

class DishService {
  async getAllDishes() {
    const response = await axiosInstance.get('/dishes');
    return response.data;
  }

  async getDishesByCategory(category) {
    const response = await axiosInstance.get(`/dishes?category=${category}`);
    return response.data;
  }

  async getDishById(id) {
    const response = await axiosInstance.get(`/dishes/${id}`);
    return response.data;
  }

  async createDish(dishData) {
    const response = await axiosInstance.post('/dishes', dishData);
    return response.data;
  }

  async updateDish(id, dishData) {
    const response = await axiosInstance.put(`/dishes/${id}`, dishData);
    return response.data;
  }

  async deleteDish(id) {
    const response = await axiosInstance.delete(`/dishes/${id}`);
    return response.data;
  }
}

export default new DishService();